www-src/app/replicator/replicator_backups.coffee

```diff
@@ -227,19 +227,40 @@ module.exports =
     uploadPicture: (path, device, callback) ->
         fs.getFileFromPath path, (err, file) =>
             return callback err if err
+            @createFile file, path, device, (err, res, body) =>
+                return callback err if err
+                @createBinary file, body._id, (err) =>
+                    return callback err if err
+                    @createPhoto path, callback
+
+
+    createBinary: (file, fileId, callback) ->
+        # Standard Blob isn't available on android prior to 4.3 ,
+        # and FormData doesn't work on 4.0 , so we use FileTransfert plugin.
+        if device.version? and device.version < '4.3'
+            @createBinaryWFiltTransfert file, fileId, callback
+
+        else
             fs.getFileAsBlob file, (err, content) =>
                 return callback err if err
+                @createBinaryWFormData content, fileId, callback
 
-                @createFile file, path, device, (err, res, body) =>
-                    return callback err if err
 
-                    @createBinary content, body._id, (err, success) =>
-                        return callback err if err
+    createBinaryWFiltTransfert: (file, fileId, callback) ->
+        options = @config.makeDSUrl("/data/#{fileId}/binaries/")
+        options.fileName = 'file'
+        options.mimeType = file.type
+        options.headers =
+            'Authorization': 'Basic ' +
+                             btoa(@config.get('deviceName') + ':' +
```

Comment:
Yes, but fixed (upstream) in https://github.com/cozy/cozy-mobile/pull/141 , avoiding special chars in device name. 
But you are right, it's cleaner to ` unescape(encodeURIComponent(( ` here too. I will do this in  https://github.com/cozy/cozy-mobile/pull/141 .